### ISSUE TITLE
Wrong access token in the websocket call.

### ISSUE THREAD

**Issue by OndraBasler** [lepicekmichal/SignalRKore#8]
So far I have used Android SDK for SignalR. I want to migrate to your solution so I can create the KMM app. The Android SDK works as expected. I have a problem with authentication to the web socket. I have this scenario:

### Scenario
As far as I understand it, it is the case with the negotiation redirection.

#### 1. Authenticate with web server:
POST https://{WebServerUrl}/api/players/login
Returns access token for negotiation

#### 2. Negotiate with web server:
POST https://{WebServerUrl}/hubs/players/negotiate?negotiateVersion=1
Bearer token from login response. The JSON response contains fields:
”url” - url for Azure SignalR service (you need to use full url as it is returned, also with query params)
”accessToken” - access token for Azure SignalR service

#### 3. Negotiate with Azure SignalR service:
POST https://{url} (full url returned in previous request, also with query params)
Bearer token from previous response
this returns JSON response that contains fields:
”connectionId” and "connectionToken" - parameters for wss connection.

#### 4. Create websocket connection with Azure SignalR service
wss://{url} (full url returned in previous request, also with query params)
additonal query params are used:
**Authorization header - access token from 2nd negotiate response** - this header is wrong
id query parameter - connectionToken from 2nd negotiate response

### Code

In the nutshell I use your SDK as follows:
```kotlin
val connection = HubConnectionBuilder.create(configuration.baseUrl) {
    accessToken = getSignalrJwtToken()
}
connection.start()
on(methodName, clazz).collect { message ->
    // do something with a message
}
```

### Issue
There are three calls that uses an access token in the header:
2. Initial negotiation call
3. Redirected negotiation call
4. Web socket call.
The initial negotiation call should use the access token provided to SDK and the redirected negotiation call and web socket call should use the access token returned from the the initial negotiation call.

Current behavior:
2. and 4. calls use the access token passed to SDK. The 4. websocket call fails with 401 and www-authenticate Bearer error="invalid_token", error_description="The signature key was not found"

Expected behavior:
The 2. call uses the access token passed to SDK and 3. and 4. calls use the access token received from the 2. call.

**Comment by lepicekmichal** [comment]
Thank you for a detailed report. I have published a new version, 0.8.7. Please check if it is working for you now.

**Comment by OndraBasler** [comment]
Thank you for a swift resolution. The update works.
